migrate/repo_checks.py

```diff
@@ -78,6 +97,253 @@ def dry_run(self):
         raise NotImplementedError
 
 
+class EnsureWorkflowTemplates(Check):
+    """
+    There are certain github action workflows that we to exist on all
+    repos exactly as they are defined in the `.github` repo in the org.
+
+    Check to see if they're in a repo and if not, make a pull request
+    to add them to the repository.
+    """
+
+    def __init__(self, api: GhApi, org: str, repo: str):
+        super().__init__(api, org, repo)
+
+        self.workflow_templates = [
+            "self-assign-issue.yml",
+            "add-depr-ticket-to-depr-board.yml",
+            "commitlint.yml",
+            "add-remove-label-on-comment.yml",
+        ]
+
+        self.branch_name = "repo_checks/ensure_workflows"
+
+        self.files_to_create = []
+        self.files_to_update = []
+        self.dot_github_template_contents = {}
+
+    def is_relevant(self):
+        return is_public(self.api, self.org_name, self.repo_name)
+
+    def check(self):
+        """
+        See if our workflow templates are in the repo and have the same content
+        as the default templates in the `.github` repo.
+        """
+        # Get the current default branch.
+        repo = self.api.repos.get(self.org_name, self.repo_name)
+        default_branch = repo.default_branch
+
+        files_that_differ, files_that_are_missing = self._check_branch(default_branch)
+        # Return False and save the list of files that need to be updated.
+        if files_that_differ or files_that_are_missing:
+            self.files_to_create = files_that_are_missing
+            self.files_to_update = files_that_differ
+            return (
+                False,
+                f"Some workflows in this repo don't match the template.\n\t\t{files_that_differ=}\n\t\t{files_that_are_missing=}",
+            )
+
+        return (
+            True,
+            "All desired workflows are in sync with what's in the .github repo.",
+        )
+
+    def _check_branch(self, branch_name):
+        """
+        Check the contents the listed workflow files on a branch against the
+        default content in the .github folder.
+        """
+        dot_github_default_branch = self.api.repos.get(
+            self.org_name, ".github"
+        ).default_branch
+        # Get the content of the .github files, maybe this should be a memoized
+        # function since we'll want to get the same .github content from all
+        # the repos.
```

Comment:
I see `@cache` on top of `get_github_file_contents`, which seems like the right thing to do 👍🏻 Should we delete this comment, then?
```suggestion
```